### ISSUE TITLE
Assert FsCheck with Unquote instead of xunit.assert

### ISSUE THREAD

**Issue by ploeh** [fscheck/FsCheck#160]
The *FsCheck.Test* project has a reference to [xunit.assert](https://www.nuget.org/packages/xunit.assert), and uses the xUnit.net assertion library to express various assertions.

xunit.assert isn't particularly F#-friendly, and has some (in my opinion) unfortunate implicit behaviour; [here's an example](https://xunit.codeplex.com/workitem/9856) - even though that particular issue was marked as *closed*, it was never properly addressed.

When expressing assertions in F#, I find [Unquote](https://www.nuget.org/packages/Unquote) much more suitable. Would it be OK to remove xunit.assert and rewrite the assertions with Unquote?

There are some 34 assertions that currently depend on xunit.assert. I'll be happy to do this work in a pull request, if you agree that this would be a good idea.

**Comment by ploeh** [comment]
A couple of questions regarding this has come up:

I can't figure out exactly how *FsCheck.Test* references *xunit.assert*. If I ask the NuGet package manager console to report the packages for the project, I only get this:

```
PM> Get-Package -ProjectName FsCheck.Test

Id                                  Versions                                 ProjectName
--                                  --------                                 -----------
xunit.runner.visualstudio           {2.1.0}                                  FsCheck.Test
```

Still, I can see in the project's references that *xunit.assert* has this `HintPath`: `..\..\packages\xunit.assert\lib\portable-net45+win8+wp8+wpa81\xunit.assert.dll`, so it looks like it's a package of some sort.

It's not included in `paket.references` either, so it's not clear to me exactly how this dependency is managed.

Still, I attempted to remove the dependency simply by deleting the reference from Visual Studio. As expected, this causes *FsCheck.Test* to fail compilation, but it also causes the following projects to fail compilation:

- FsCheck.CSharpExamples
- FsCheck.MsTest.Examples
- FsCheck.NUnit.CSharpExamples

Why does this happen?

As far as I can tell, none of these projects depend on *FsCheck.Test*, so it's not clear to me why this happens.

**Comment by kurtschelfthout** [comment]
I'd love to convert to unquote. 

The behaviour you describe is magical. Though I can explain why nuget only finds the one package: paket has to do all kinds of special stuff for it because it is a vs test runner. Don't expect nuget to see the others.

**Comment by ploeh** [comment]
What's the correct way to remove the *xunit.assert* reference, and add the *Unquote* reference?

Why does compilation fail for those unrelated projects?

**Comment by kurtschelfthout** [comment]
The correct way to add and remove references is to use paket remove/add from/to the specified project. In FsCheck.Test's paket.references file, there is now a dependency on xunit. I think that needs to be removed and you need to add only the dependencies you need. (Before 2.0, they hadn't separated all these different parts, so doing so was essentially impossible. This is just a leftover of that)

I had a look at your other issue. I can reproduce that behavior, but interestingly it's only intellisense that has a problem with those projects. They actualy build fine (have a look at the build output window to verify this). So my guess is that the particular way in which paket adds references, doesn't play well with the add/remove reference from visual studio and that confuses intellisense. Have a look at the change you actually made in a git diff to understand this.

All in all, I'm not so worried about this. Just use paket to remove xunit dependency from FsCheck.Test first, then use paket add to add the dependency you do want, that should work.

**Comment by ploeh** [comment]
My experience with Paket is quite limited, so thank you for explaining this to me. I do, however, have problems making this work:

```
$ .paket/paket remove nuget xunit.assert project FsCheck.Test
Paket version 2.8.2.0
Package xunit.assert was not installed in project FsCheck.Test.fsproj in group Main
C:\Users\mark\Documents\FsCheck\paket.dependencies doesn't contain package xunit.assert in group Main. ==> Ignored
Dependencies files saved to C:\Users\mark\Documents\FsCheck\paket.dependencies
...
```

Nothing changes inside of Visual Studio either. *xunit.assert* is still referenced from *FsCheck.Test*. What am I doing wrong?

**Comment by kurtschelfthout** [comment]
As usual, the program is pedantically right: `xunit.assert` is not installed. `xunit` is installed.

Paket does not count transitive dependencies as installed (that's one of the things that makes it a saner package manager, in my opinion - although in this particular case it may be confusing you :) )

In any case, `paket remove nuget xunit project FsCheck.Test` should work.

And then you can install the xunit packages you actually need, with say:

`paket add nuget xunit.core project FsCheck.Test`

You may need more than that but you get the idea.

**Comment by ploeh** [comment]
Resolved in #162
